Modules/_posixsubprocess.c

```diff
@@ -724,9 +724,9 @@ do_fork_exec(char *const exec_array[],
              int errpipe_read, int errpipe_write,
              int close_fds, int restore_signals,
              int call_setsid, pid_t pgid_to_set,
-             int call_setgid, gid_t gid,
-             int call_setgroups, size_t groups_size, const gid_t *groups,
-             int call_setuid, uid_t uid, int child_umask,
+             gid_t gid,
+             size_t groups_size, const gid_t *groups,
```

Comment:
change groups_size to a Py_ssize_t as that is what we pass in. We can then use it being < 0 to mean the former `!call_setgroups`.  A call to C's `setgroups(0, NULL)` via an empty list of groups has meaning.